### PR TITLE
add binary lex

### PR DESCRIPTION
**Pull request by alejandrolechuga**
-- I'll probably need re-create this commit from my home computer ... idk why my other account shows my other avatar after specifying my github user.

### CODE REVIEW

File: lexer/src/lib.rs [hunk 1]
```diff
@@ -124,45 +125,60 @@ where I: Iterator<Item = char>,
...
                         self.stream.push_while(&mut s, |c| is_ascii_octaldigit(c));
                         return Ok(TokenType::Number(s))
                     },
+                    // binary part
+                    Some(c) if c == 'b' || c == 'B' => {
+                        self.stream.skip_char();
+                        s.push(c);
+
+                        match self.stream.next() {
+                            // if is a digit between 0 and 7
```
Comment (alejandrolechuga) [inline lexer/src/lib.rs:171]: comment should be for binary

File: lexer/src/charclass.rs [hunk 2]
```diff
@@ -54,4 +54,11 @@ pub fn is_ascii_octaldigit(c: char) -> bool {
         Some(_) => true,
         None => false
     }
-}
\ No newline at end of file
+}
+
+pub fn is_ascii_binarydigit(c: char) -> bool {
+    match c.to_digit(2) {
```
Comment (delta62) [inline lexer/src/charclass.rs]: We can do `c.to_digit(2).is_some()`

**Review by delta62 — APPROVED** [review]
LGTM!

**Comment by delta62** [comment]
@alejandrolechuga I think it's doing it by the email address in your `~/.gitconfig`?

**Pull request merged**